standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3342,8 +3342,8 @@ public List<String> listPartitionNames(final String catName, final String dbName
     final boolean isEmptyFilter = exprBytes.length == 1 && exprBytes[0] == -1;
     ExpressionTree tmp = null;
     if (!isEmptyFilter) {
-      tmp = PartFilterExprUtil.makeExpressionTree(expressionProxy, exprBytes,
-          getDefaultPartitionName(defaultPartName), conf);
+      String exprString = PartFilterExprUtil.getExpressionString(expressionProxy, exprBytes, getDefaultPartitionName(defaultPartName), conf);
+      ExpressionTree exprTree = ExpressionTree.fromString(exprString);
     }
     final ExpressionTree exprTree = tmp;
```

Comment:
you can remove this as well as it is used further down in 
**final ExpressionTree exprTree = exprString.isEmpty() ? ExpressionTree.EMPTY_TREE : ExpressionTree.fromString(exprString);**